### PR TITLE
VIH-8181 Quick Link participant name Special Char validation

### PR DESCRIPTION
**Pull request by alberto-hortiguela**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/VIH-8181

### Change description ###
Added Participant Quick-link Name validation (back-end) for special chars
Unit Test updated

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: VideoApi/VideoApi/Validations/AddQuickLinkParticipantRequestValidation.cs [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
 
 namespace VideoApi.Validations
 {
     public class AddQuickLinkParticipantRequestValidation : AbstractValidator<AddQuickLinkParticipantRequest>
     {
         public static readonly string NoNameErrorMessage = "Name is required";
         public static readonly string NoUserRoleErrorMessage = "UserRole is required";
+        public static readonly string SpecialCharNameErrorMessage = @"Name must not contain the following characters ! ” # $ % & ( ) * + , . / : ; < = > ? @ [ \ ] ^ _ ` { | } ~";
+
+        public static readonly Regex RegexName = new Regex(@"^[^!”#$%&()*+,./:;<=>?@[\\\]^_`{|}~]+$");
```
Comment (omio-ma) [inline VideoApi/VideoApi/Validations/AddQuickLinkParticipantRequestValidation.cs:13]: are we using this property anywhere, from what I can see it only appears to be used here, if that is the case can we change this to a string as we are converting this back to string?

Comment (alberto-hortiguela) [reply VideoApi/VideoApi/Validations/AddQuickLinkParticipantRequestValidation.cs]: done now

File: VideoApi/VideoApi/Validations/AddQuickLinkParticipantRequestValidation.cs [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
         public static readonly string NoNameErrorMessage = "Name is required";
         public static readonly string NoUserRoleErrorMessage = "UserRole is required";
+        public static readonly string SpecialCharNameErrorMessage = @"Name must not contain the following characters ! ” # $ % & ( ) * + , . / : ; < = > ? @ [ \ ] ^ _ ` { | } ~";
+
+        public static readonly string RegexName = new Regex(@"^[^!”#$%&()*+,./:;<=>?@[\\\]^_`{|}~]+$").ToString();
 
         public AddQuickLinkParticipantRequestValidation()
         {
             RuleFor(x => x.Name).NotEmpty().WithMessage(NoNameErrorMessage);
+            RuleFor(x => x.Name).Matches(RegexName).WithMessage(SpecialCharNameErrorMessage).When(x => !string.IsNullOrEmpty(x.Name) ); ;
```
Comment (krishnapottekkat-hmcts) [inline VideoApi/VideoApi/Validations/AddQuickLinkParticipantRequestValidation.cs:18]: Remove the space and double semicolon ?

Comment (alberto-hortiguela) [reply VideoApi/VideoApi/Validations/AddQuickLinkParticipantRequestValidation.cs]: thats´s done now

**Review by omio-ma — APPROVED** [review]

**Review by krishnapottekkat-hmcts — APPROVED** [review]

**Pull request merged**